crypto.go

```diff
@@ -26,12 +25,16 @@ func getKey(key string) []byte {
 	}
 	return decodedkey
 }
-func encryptDocs(hits gjson.Result, stream cipher.Stream, fr io.Writer) {
-	var res strings.Builder
-	fmt.Fprintf(&res, "%s\n", hits.Raw)
-	plainText := []byte(res.String())
-	cipherText := make([]byte, len(plainText))
-	stream.XORKeyStream(cipherText, plainText)
+
+func encryptDocs(buf *bytes.Buffer, stream cipher.Stream, fr io.Writer) {
```

Comment:
Can this take a io.Reader  instead of bytes.Buffer?

Reply:
Potentially, although I see no immediate benefit from it.